tests/integration/dhcp_test.py

```diff
@@ -192,6 +192,126 @@ def test_dhcp_for_bond_with_ip_address_and_slave(dhcp_env,
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv4']['enabled'] = True
+    dhcp_cli_desired_state['ipv4']['dhcp'] = True
+    dhcp_cli_desired_state['ipv4']['auto-dns'] = False
+    dhcp_cli_desired_state['ipv6']['enabled'] = True
+
+    netapplier.apply(desired_state)
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support DNS query, need to make
+    # sure dhcp DNS entries got ignored.
+
+
+def test_ipv4_dhcp_ignore_routes(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv4']['enabled'] = True
+    dhcp_cli_desired_state['ipv4']['dhcp'] = True
+    dhcp_cli_desired_state['ipv4']['auto-routes'] = False
+    dhcp_cli_desired_state['ipv6']['enabled'] = True
+
+    netapplier.apply(desired_state)
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support route query, need to make
+    # sure dhcp route got ignored. And this also require change DHCP server
+    # from dnsmasq to dhcpcd to pushing extra routes.
+
+
+def test_ipv6_dhcp_and_autoconf_ignore_gateway(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv6']['dhcp'] = True
+    dhcp_cli_desired_state['ipv6']['autoconf'] = True
+    dhcp_cli_desired_state['ipv6']['auto-gateway'] = False
+
+    netapplier.apply(desired_state)
+
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support route query, need to make
+    # sure DHCP gateway got ignored.
+
+
+def test_ipv6_dhcp_and_autoconf_ignore_dns(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv6']['dhcp'] = True
+    dhcp_cli_desired_state['ipv6']['autoconf'] = True
+    dhcp_cli_desired_state['ipv6']['auto-dns'] = False
+
+    netapplier.apply(desired_state)
+
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support DNS query, need to make
+    # sure DHCP DNS entires got ignored.
+
+
+def test_ipv6_dhcp_and_autoconf_ignore_routes(dhcp_env):
+    desired_state = statelib.show_only((DHCP_CLI_NIC,))
+    dhcp_cli_desired_state = desired_state[INTERFACES][0]
+    dhcp_cli_desired_state['state'] = 'up'
+    dhcp_cli_desired_state['ipv6']['dhcp'] = True
+    dhcp_cli_desired_state['ipv6']['autoconf'] = True
+    dhcp_cli_desired_state['ipv6']['auto-routes'] = False
+
+    netapplier.apply(desired_state)
+
+    assertlib.assert_state(desired_state)
+    # TODO(Gris Ge): Once libnmstate support route query, need to make
+    # sure dhcp route got ignored. And this also require change DHCP server
+    # from dnsmasq to dhcpcd to pushing extra routes.
```

Comment:
You can push extra routes with ` --dhcp-option=option:classless-static-route,198.51.100.72/30,192.0.2.23` in dnsmasq.

Reply:
This seems to be addressed, so the TODO can be removed, can it not?